### PR TITLE
RC69: crash fixes

### PR DESCRIPTION
**Pull request by AndrewMeadows**
https://highfidelity.fogbugz.com/f/cases/15968/crash-when-accessing-an-Animation-via-JS-before-it-has-loaded

* fix a crash when accessing an Animation from JS before it was loaded
* fix a crash when computing inertia tensor on RigidBody with triangle-mesh shape
* add protection against a few theoretical crash modes

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
**Test Plan:**

**(1)** Using interface client from `hifi/master` run the script below: it should crash the interface right away:
```
// crash.js
var animation = AnimationCache.getAnimation("http://www.fungibleinsight.com/faces/gangnam_style_2.fbx");
var names = animation.jointNames;
```
**(2)** Repeat step **(1)** but using the interface client from this PR: it should NOT crash.

**(3)** There is a slightly distinct crash mode in the script below that is also fixed in this PR:
```
// crash-again.js
var animation = AnimationCache.getAnimation("http://www.fungibleinsight.com/faces/gangnam_style_2.fbx");
var frames = animation.frames;
```
**(4)** Make sure the main menus and their sub-menus show up when you select them.

**(5)** Make any dynamic object: verify it tumbles and collides correctly.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10624/HighFidelity-Beta-PR13391-4880492.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10624/HighFidelity-Beta-PR13391-4880492.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10624/HighFidelity-Beta-Interface-PR13391-4880492.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10624/HighFidelity-Beta-PR13391-4880492.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10624/HighFidelity-Beta-Interface-PR13391-4880492.exe

File: libraries/ui/src/ui/Menu.cpp [hunk 1]
```diff
@@ -402,8 +402,10 @@ MenuWrapper* Menu::addMenu(const QString& menuName, const QString& grouping) {
 
     // hook our show/hide for popup menus, so we can keep track of whether or not one
     // of our submenus is currently showing.
-    connect(menu->_realMenu, &QMenu::aboutToShow, []() { _isSomeSubmenuShown = true; });
-    connect(menu->_realMenu, &QMenu::aboutToHide, []() { _isSomeSubmenuShown = false; });
+    if (menu && menu->_realMenu) {
+        connect(menu->_realMenu, &QMenu::aboutToShow, []() { _isSomeSubmenuShown = true; });
+        connect(menu->_realMenu, &QMenu::aboutToHide, []() { _isSomeSubmenuShown = false; });
```
[the remark this answers is not in the document]
Comment (AndrewMeadows) [reply libraries/ui/src/ui/Menu.cpp]: @Atlante45 could  you please elaborate on what you mean by "add some context objects"?

[the remark this answers is not in the document]
Comment (Atlante45) [reply libraries/ui/src/ui/Menu.cpp]: (talked in person)

**Comment by conklin94122** [comment]
@AndrewMeadows If this is QA Ready please label it. If it's not, never mind :)

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Interface does not crash.

**Pull request merged**